File: output/deepseek_r1/avatar/algo_based_translation/python/java/atcoder_ABC123_C.java

import java.util.Scanner;

public class atcoder_ABC123_C {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] a = new int[5];
        for (int i = 0; i < 5; i++) {
            a[i] = scanner.nextInt();
        }
        int minVal = a[0];
        for (int num : a) {
            if (num < minVal) {
                minVal = num;
            }
        }
        int result = (int) Math.ceil((double) n / minVal) + 4;
        System.out.println(result);
    }
}

File: output/deepseek_r1/avatar/algo_based_translation/java/python/atcoder_ABC123_C.py

import math

def main():
    N = int(input())
    min_val = 0
    for i in range(5):
        if i == 0:
            min_val = int(input())
        else:
            num = int(input())
            if num < min_val:
                min_val = num
    result = math.ceil(N / min_val) + 4
    print(result)

if __name__ == "__main__":
    main()

